src/MemoryRouter.test.tsx

```diff
@@ -315,6 +315,24 @@ describe("MemoryRouter", () => {
         });
       });
 
+      it("correctly uses default as param", async () => {
+        memoryRouter.setCurrentUrl("/path?queryParam=123");
+        expect(memoryRouter).toMatchObject({
+          asPath: "/path?queryParam=123",
+          pathname: "/path",
+          query: { queryParam: "123" },
+        });
+      });
+
+      it("supports as param", async () => {
+        memoryRouter.setCurrentUrl("/path?queryParam=123", "/path");
+        expect(memoryRouter).toMatchObject({
+          asPath: "/path",
+          pathname: "/path",
+          query: { queryParam: "123" },
+        });
+      });
```

Comment:
I'd like to include a few more tests, with `as` params like:
- `"/path?foo=111"`
- `"/other/path#hash"`
- `{ pathname: "/path", query: { foo: 111 } }`

Reply:
On it :) We will add some it.each() test to cover a variety of cases, both with the `as` param as a string as well as an object